### DISCUSSION TITLE
Versioning

### DISCUSSION THREAD

**Discussion by ericcornelissen** [donoevil-oss/donoevil-oss.github.io#5] [Ideas]
Currently the website reads: "_[W]henever I link to this site from a specific repository I am pledging that [...]_". However, what if I agree with the pledge as it is today but not after some change in the future? I may want to pledge to a specific version instead.

Granted, if the versions are centralized on this website then this wouldn't be a foolproof solution. But at least if there are versions users can come up with ways to enshrine the text of a version in an appropriate location (e.g. a file in their repo or their [GitHub profile README](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/managing-your-profile-readme)).

As an example, the [Contributor Covenant ("A Code of Conduct for Open Source Communities")](https://www.contributor-covenant.org/) also has multiple versions. If there's a new version users can review the changes before adopting it.

As an aside, the version number _could_ also be part of the badge (ref. https://github.com/donoevil-oss/donoevil-oss.github.io/issues/2).

**Comment by psiinon** [comment]
Good point, and thanks for the example :) 
Do you think we should do exactly the same?
Right now I'm still trying to work out if this idea has enough support to be viable...

**Reply by ericcornelissen** [reply]
> Do you think we should do exactly the same?

Can you elaborate, are you referring to _"Contributor Covenant uses semantic versioning for its revisions, so all URLs are permanent"_ from the homepage?
If so, I don't have a strong opinion to using something like semantic versioning or simple incremental versioning (i.e. just v1, v2, v3, etc.) for non-software things.

> Right now I'm still trying to work out if this idea has enough support to be viable...

Of course :slightly_smiling_face: I don't advocate starting versioning until you (we?) have something you're happy with.
However, to get any real support I think having versions is crucial. I feel like I can't really pledge to something if it can arbitrarily change, nor can I take anyone taking the pledge serious for the same reason. Versioning partly solves this problem by allowing changes to the pledge without it affecting "users" (i.e. those that took the pledge in the past).
